### DISCUSSION TITLE
Dump / Import postgresql from Digital Ocean droplet (One-click install)

### DISCUSSION THREAD

**Discussion by florianjs** [directus/directus#6651] [Archive]
Hey ! 🙌 

I'm looking for some help 😊
I've created some collections / Data models into Directus, and I want to Open Source the Database as a model for others.

The thing is ... I'm a complete noob at postgresql 🙃 Mostly a Front-end dev - Do you guys have some tutos / helpful content to :
1 - Export the Database (Only the data models /  Collection items) from DO Droplet
2 - Import the database into DO Droplet

Hopefully, someone in the community already did that ? 🥳

Thank you very much, and have a nice day !

**Comment by florianjs** [comment]
Ok, so I've found a "solution" using TablePlus : https://www.digitalocean.com/community/questions/how-do-you-remote-connect-with-pgadmin-4-to-a-postgresql-on-a-droplet (could help someone) 

1. Connect via SSH ssh -L 63333:localhost:5432 user@111.111.111.111
2. Add your DB on TablePlus host : localhost port : 63333 user : directus password : DB_password

Then, once I'm connect, I tried the following : 
Backup the DB and then Restore on an another Directus => Don't work ?
Export DB => Import on the other instance, but I need to Drop the tables first. When I do that, I got an error after a npx directus start when I try to login : "duplicate value detected"


EDIT : Okay. So I have dumped all the datas in "directus_activity" table and..... it's working. Don't really know why 🙃

**Reply by rijkvanzanten** [reply]
> When I do that, I got an error after a npx directus start when I try to login : "duplicate value detected"

TablePlus doesn't reset the identify sequence after a dump import, so you'll have to add a SETVAL query to reset that, for example:

```sql
SELECT SETVAL('public.directus_activity_id_seq', COALESCE(MAX(id), 1) ) FROM public.directus_activity;
```

**Reply by florianjs** [reply]
Good to know ! Thank you @rijkvanzanten

**Reply by Pachat** [reply]
I had the same problem that was solved the same way. Thank you @rijkvanzanten

It looks like this has to be done for any collection that has an ID which is an INT.

For PostgreSQL:
```
DO $$ 
DECLARE
    collection text;
BEGIN
    FOR collection IN (
        SELECT table_name
        FROM information_schema.columns
        WHERE table_schema = 'public' AND column_name = 'id' AND data_type = 'integer'
    )
    LOOP
        EXECUTE format('SELECT SETVAL(''public.%I_id_seq'', COALESCE(MAX(id), 1)) FROM public.%I;', collection, collection);
    END LOOP;
END $$;
```
assuming integer primary key columns are called `id`

**Reply by huantt** [reply]
> ```sql
> ```sql
> SELECT SETVAL('public.directus_activity_id_seq', COALESCE(MAX(id), 1) ) FROM public.directus_activity;
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
>     
> ```

Saved my hours, thank you!!!
